### DISCUSSION TITLE
PAGView和PAGImageView在Android上崩溃

### DISCUSSION THREAD

**Discussion by Mr-Dimple** [Tencent/libpag#2326] [Q&A] [answered]
【版本信息】
PAG SDK 版本 4.3.33
【平台信息】
Android原生
【问题描述】
我们在线上遇到了这个崩溃：PAGView和PAGImageView都有，但是PAGVIew少一些
java.lang.UnsatisfiedLinkError: No implementation found for void org.libpag.PAGAnimator.nativeInit() (tried Java_org_libpag_PAGAnimator_nativeInit and Java_org_libpag_PAGAnimator_nativeInit__)
at org.libpag.PAGAnimator.nativeInit(Native Method)
at org.libpag.PAGAnimator.<clinit>(SearchBox:6)
at org.libpag.PAGAnimator.a(SearchBox:115000)
at org.libpag.PAGImageView.e(SearchBox:13)
at org.libpag.PAGImageView.<init>(SearchBox:20)

java.lang.UnsatisfiedLinkError: No implementation found for void org.libpag.PAGPlayer.nativeInit() (tried Java_org_libpag_PAGPlayer_nativeInit and Java_org_libpag_PAGPlayer_nativeInit__)
at org.libpag.PAGPlayer.nativeInit(Native Method)
at org.libpag.PAGPlayer.<clinit>(SearchBox:11)
at org.libpag.PAGView.b(SearchBox:12)
at org.libpag.PAGView.<init>(SearchBox:8)

辛苦帮忙看看是什么原因，我们应该怎么改

**Comment by kevingpqi123** [comment]
1、应该是架构没有匹配上，你们可以分析下出现类似 crash 问题的手机系统架构，然后做下匹配
2、你们使用的版本比较老，建议升级至 PAG 4.3 的最新版本

**Reply by Mr-Dimple** [reply]
架构没匹配上是指？arm64？出现问题的机型很多，我们应该在什么地方做下适配呢
目前我们集成的版本是4.3.3版本 官网上最新的版本是4.3.57 看了下中间的改动 看上去也没有相关的改动。

**Reply by kevingpqi123** [reply, marked as answer]
pag 的 aar 文件默认支持的是 arm64-v8a、armeabi、armeabi-v7a 三个架构，如果你们使用的是 aar，首先看下架构匹配的问题；如果是 so 下发，确保使用之前 so 被正确加载
还有就是注意下混淆规则问题：
    -keep class org.libpag.** {*;}
    -keep class androidx.exifinterface.** {*;}

主要是以上原因，PAG 4.3 版本目前使用的业务很多，暂时还没有其它业务反馈过代码层面的这类问题

**Reply by Mr-Dimple** [reply]
我们使用的是aar的方式接入的，线上崩溃的机型中有华为P30，p30的架构是arm64-v8a的，如果是架构不匹配应该崩溃的量级会很大，但目前看都是偶现的，有没有什么日志能帮助排查具体是什么原因

**Reply by kevingpqi123** [reply]
目前的堆栈有效信息不是太多，其它业务也没有反馈过类似堆栈，建议你们先升级到最新版本再看下，同时建议你们分析下复现路径，看看测试同学能否复现到，如果有复现路径，分析定位也可以快很多

**Reply by cai123** [reply]
遇见少量用户重复上报类似问题，用的是4.3.33版本 @Mr-Dimple 请问你们的问题解决了吗

**Reply by LXD312569496** [reply]
大佬们问题解决了吗
